src/dbnode/storage/namespace/types.go

```diff
@@ -104,6 +109,15 @@ type IndexOptions interface {
 	BlockSize() time.Duration
 }
 
+// Schema represents namespace schema
+type Schema interface {
+	Version() uint32
+	Equal(Schema) bool
+	Get() *desc.MessageDescriptor
```

Comment:
Not a fan of directly exposing a dependency on a library's exported type here if we can help it. 

How would you feel about exposing types corresponding to the generic [protobuf descriptor](https://github.com/protocolbuffers/protobuf/blob/199d82fde1734ab5bc931cd0de93309e50cd7ab9/src/google/protobuf/descriptor.proto), or the `.proto` bytes instead?

Reply:
The .proto bytes aren't useful to any other parts of the code as they'll have to re-implement all the parsing code and error checking so lets not do that.

Not sure what you mean about the generic protobuf descriptor since you just linked to a .proto text file?

Regardless of what we decide we should settle on something thats actually usable by the rest of the code-base even if that means we have to expose a dependency here (we could wrap it in a type alias or struct or something so that the dependency is only referenced from one package and then if we ever need to change it we can just change it there)

Reply:
ok, I will alias it.

Reply:
> ok, I will alias it.

forget golang has no alias, it is another type all together. I don't want user to have to assert it back to  *desc.MessageDescriptor everything they uses it.